File: backend/app/schemas/families.py
"""Family and family member schemas."""

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, Field

from ..models.family import FamilyMemberRole


class FamilyCreateSchema(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)


class FamilySchema(BaseModel):
    id: str
    name: str
    created_by: str
    created_at: datetime

    class Config:
        from_attributes = True


class FamilyMemberCreateSchema(BaseModel):
    display_name: str = Field(..., min_length=1, max_length=255)
    date_of_birth: Optional[date] = None
    sex: Optional[str] = Field(None, max_length=20)
    role: FamilyMemberRole = FamilyMemberRole.MEMBER
    user_id: Optional[str] = None


class FamilyMemberSchema(BaseModel):
    id: str
    family_id: str
    user_id: Optional[str]
    display_name: str
    date_of_birth: Optional[date]
    sex: Optional[str]
    role: FamilyMemberRole
    joined_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/families.ts
import apiClient from './client'

export interface Family {
  id: string
  name: string
  created_by: string
  created_at: string
}

export interface FamilyMember {
  id: string
  family_id: string
  user_id: string | null
  display_name: string
  date_of_birth: string | null
  sex: string | null
  role: 'admin' | 'member' | 'viewer'
  joined_at: string
}

export const familiesAPI = {
  listFamilies: async (): Promise<Family[]> => {
    const response = await apiClient.get('/families')
    return response.data
  },

  createFamily: async (name: string): Promise<Family> => {
    const response = await apiClient.post('/families', { name })
    return response.data
  },

  listMembers: async (familyId: string): Promise<FamilyMember[]> => {
    const response = await apiClient.get(`/families/${familyId}/members`)
    return response.data
  },

  addMember: async (
    familyId: string,
    payload: { display_name: string; email?: string; date_of_birth?: string; sex?: string; role?: string }
  ): Promise<FamilyMember> => {
    const response = await apiClient.post(`/families/${familyId}/members`, payload)
    return response.data
  },
}
